plugins/amongus/amongus.cpp

```diff
@@ -5,122 +5,132 @@
-	stream << "Client ID: " << std::to_string(fields.clientId) << '\n';
-	stream << "Player ID: " << std::to_string(playerFields.playerId) << '\n';
-	stream << "Color ID: " << std::to_string(playerFields.colorId) << '\n';
-	stream << "Skin ID: " << std::to_string(playerFields.skinId) << '\n';
-	stream << "Hat ID: " << std::to_string(playerFields.hatId) << '\n';
-	stream << "Pet ID: " << std::to_string(playerFields.petId) << '\n';
-	stream << "Dead: " << (playerFields.isDead ? "true" : "false") << '\n';
-	stream << "Host ID: " << std::to_string(fields.hostId) << '\n';
-	stream << "Public game: " << (fields.isGamePublic ? "true" : "false");
+MumbleStringWrapper mumble_getDescription() {
+	static const char description[] = "Provides positional audio functionality for Among Us."
+									  "Context and identity are provided.";
 
-	identity = utf8ToUtf16(stream.str());
+	MumbleStringWrapper wrapper;
+	wrapper.data           = description;
+	wrapper.size           = sizeof(description) - 1;
+	wrapper.needsReleasing = false;
 
-	return true;
+	return wrapper;
 }
 
-static int tryLock(const std::multimap< std::wstring, unsigned long long int > &pids) {
-	const std::string name = "Among Us.exe";
-	const auto id          = Process::find(name, pids);
-	if (!id) {
-		return false;
-	}
-
-	game = std::make_unique< Game >(id, name);
-	if (!game->isOk()) {
-		game.reset();
-		return false;
-	}
+uint32_t mumble_getFeatures() {
+	return MUMBLE_FEATURE_POSITIONAL;
+}
 
-	const auto fields = game->clientFields();
-	if (!inGame(fields)) {
-		game.reset();
-		return false;
+uint8_t mumble_initPositionalData(const char *const *programNames, const uint64_t *programPIDs, size_t programCount) {
+	for (size_t i = 0; i < programCount; ++i) {
+		if (strcmp(programNames[i], "Among Us.exe") != 0) {
+			continue;
+		}
+
+		game = std::make_unique< Game >(programPIDs[i], programNames[i]);
+
+		const auto error = game->init();
+		if (error == Game::NONE) {
+			const auto fields = game->clientFields();
+			if (!game->isMultiplayer(fields)) {
+				game.reset();
+				return MUMBLE_PDEC_ERROR_TEMP;
+			}
+		} else {
+			game.reset();
+			return error == Game::TEMP ? MUMBLE_PDEC_ERROR_TEMP : MUMBLE_PDEC_ERROR_PERM;
+		}
+
+		return MUMBLE_PDEC_OK;
 	}
 
-	return true;
+	return MUMBLE_PDEC_ERROR_TEMP;
 }
 
-static const std::wstring longDesc() {
-	return std::wstring(L"Supports Among Us with context and identity support.");
+void mumble_shutdownPositionalData() {
+	game.reset();
 }
 
-static std::wstring description(L"Among Us");
-static std::wstring shortName(L"Among Us");
+bool mumble_fetchPositionalData(float *avatarPos, float *avatarDir, float *avatarAxis, float *cameraPos,
+								float *cameraDir, float *cameraAxis, const char **contextPtr,
+								const char **identityPtr) {
+	const auto fields = game->clientFields();
 
-static int tryLock1() {
-	return tryLock(std::multimap< std::wstring, unsigned long long int >());
-}
+	if (!game->isMultiplayer(fields)) {
+		return false;
```

Comment:
Quoting from the docs:
> If no data can be fetched, **set all float-vectors to 0** and return false.

I recommend doing this first when entering this function especially as you have another early-return below in case the game has ended.

Reply:
Right, I just realized only the context and identity are initialized by default.

https://github.com/mumble-voip/mumble/blob/e21ad54db2004f3c09cd083b021ba42ffb631bd4/src/mumble/Plugin.cpp#L474-L504